includes/CLI/Plugin_Check_Command.php

```diff
@@ -289,6 +307,35 @@ public function list_checks( $args, $assoc_args ) {
 			)
 		);
 
+		// Filter by stability.
+		if ( ! empty( $options['stability'] ) ) {
+			$all_checks = array_filter(
+				$all_checks,
+				static function ( $item ) use ( $options ) {
+					return $options['stability'] === $item['stability'];
+				}
+			);
+		}
+
+		// Filter by categories.
+		if ( ! empty( $options['categories'] ) ) {
+			$all_checks = array_filter(
+				$all_checks,
+				static function ( $item ) use ( $options ) {
+					$categories = array_map( 'trim', explode( ',', $options['categories'] ) );
+					return array_intersect( $categories, $item['category'] );
+				}
+			);
```

Comment:
This should be simplified by reusing `Check_Categories::filter_checks_by_categories()` which is already implemented for this purpose.